### PR TITLE
main() defines things that are then used more than one place

### PR DESCRIPTION
**Pull request by boyanpenkov**
``main()`` defined ``savebib(my)`` which was then used by a few functions in ``main`` -- this may be better off with global scope, since then ``import papers`` in another script would be able to use those functions.

There's a few of these.

### CODE REVIEW

**Comment by boyanpenkov** [comment]
Confirming all tests pass.

**Comment by boyanpenkov** [comment]
Confirming I see 101 passing and 2 skipped tests.

**Comment by boyanpenkov** [comment]
Confirming I see all tests pass after the last commit.

I now understand the code better, and will pause for a bit here until I hear from @perrette  to see if this is a direction that makes sense to take.  But the exercise here is 80% me understanding the code better, which is a win either way.

**Comment by boyanpenkov** [comment]
Actually, all set with pulling all these defs out of main; nifty.

Confirming tests pass.

File: papers/extract.py [hunk 1]
```diff
@@ -120,7 +120,8 @@ def isvaliddoi(doi):
 
 
 def pdfhead(pdf, maxpages=10, minwords=200, image=False):
-    """ read pdf header
+    """
```
Comment (perrette) [inline papers/extract.py:123]: I don't mind, but I also don't see anything [here](https://peps.python.org/pep-0257) that favours one over the other.
If anything, one liner `""" read pdf header"""`

**Comment by perrette** [comment]
Great that you've been working to improving the code. I think of course it makes sense. I'd argue the the top-most priority before moving things around remains to improve test coverage -- so that we don't break things unnoticed. Test coverage is pretty poor, as I was reminded during the last few days. About moving the addcmd etc outside of main: does it make it easier for you to read / understand them? Of course the namespace separation is probably the main benefit. The reason why I had left them under main, and intertwined with the parser definition, was to have all related things in one place. The subparser feature gets pretty complex to handle in the code, and as I was writing it it was definitely better to have them inline. It may well be that it only gets confusing past this first phase -- as your modification suggests. I also had some trouble going through them now, nearly 6 years after I first wrote them.

File: papers/extract.py (hunk 1 above)
Comment (boyanpenkov) [reply papers/extract.py:123]: Yeah, you're right -- I guess I don't actually know where the fetish for 
"""
thing
"""
 came from...

**Comment by perrette** [comment]
Yeah the functions are really parts of main. However the bib.py file is more than 1300+ lines and I remember I long intended to make the main function in its separate file.py -- so that bib.py can be imported in some future application not directly related to the main cli. So I am taking this as an opportunity to do it (#43). Unless you spot some flaw or annoyance with it, I'll double down your move and accept that PR of mine. I'll take the chance to fix some bugs your PR introduced (check_install fails because parser, which lives under `main()` is not defined any more).

I agree that how config behaves is not intuitive and it will need to be improved.

**Pull request merged**